src/ucm/mmap/install.c

```diff
@@ -121,9 +157,12 @@ static ucs_status_t ucs_mmap_install_reloc(int events)
     ucm_mmap_func_t *entry;
     ucs_status_t status;
 
-    if (!ucm_global_opts.enable_mmap_reloc) {
-        ucm_debug("installing mmap relocations is disabled by configuration");
+    if (ucm_global_opts.mmap_hook_mode == UCM_MMAP_HOOK_NONE) {
+        ucm_debug("installing mmap hooks is disabled by configuration");
         return UCS_ERR_UNSUPPORTED;
+    } else if (RUNNING_ON_VALGRIND && (ucm_global_opts.mmap_hook_mode == UCM_MMAP_HOOK_BISTRO)) {
+        ucm_debug("MMAP hook mode bistro is not supported on valgrind, force reloc mode");
+        ucm_global_opts.mmap_hook_mode = UCM_MMAP_HOOK_RELOC;
```

Comment:
maybe have some inline function wrapper (e.g ucm_mmap_hook_mode()) to return the mode, adjusted when running with valgrind, and use it instead of global var? seems a bit ugly to modify the config like that